public/js/controllers/charts_controller.js

```diff
@@ -537,35 +596,40 @@ export default class extends Controller {
   }
 
   async selectChart () {
+    this.setVisibility()
     var selection = this.settings.chart = this.chartSelectTarget.value
     this.chartWrapperTarget.classList.add('loading')
-    if (isScaleDisabled(selection)) {
-      this.scaleSelectorTarget.classList.add('d-hide')
-      this.vSelectorTarget.classList.remove('d-hide')
+    this.toggleAllControls(isAllControlsDisabled(selection))
+    if (isAllControlsDisabled(selection)) {
+      this.customAxisOptionTarget.innerHTML = customXLabel(selection)
     } else {
-      this.scaleSelectorTarget.classList.remove('d-hide')
-      this.vSelectorTarget.classList.add('d-hide')
+      this.toggleBinControls(isBinDisabled(selection))
+      this.toggleScaleControls(isScaleDisabled(selection))
     }
+
     if (selectedChart !== selection || this.settings.bin !== this.selectedBin() ||
-      this.settings.axis !== this.selectedAxis()) {
-      let url = '/api/chart/' + selection
-      if (usesWindowUnits(selection)) {
-        this.binSelectorTarget.classList.add('d-hide')
-      } else {
-        this.binSelectorTarget.classList.remove('d-hide')
+      (this.settings.axis !== this.selectedAxis() && this.selectedAxis()) ||
+      isAllControlsDisabled(selection)) {
+      let searchParams = new URLSearchParams()
```

Comment:
I don't think that `URLSearchParams` works in IE. You could use the `Url` class from  `url-parse`.  We are already using the module in other places. https://www.npmjs.com/package/url-parse